### ISSUE TITLE
TSQL: Incorrect indent in if block

### ISSUE THREAD

**Issue by fdw** [sqlfluff/sqlfluff#1555]
Example code:
```sql
if @debug = 1
  begin
    print '<something>';
    print '<something else>';
    return
  end
```

## Expected Behaviour
The indentation is correct like this.

## Observed Behaviour
The third line (`print '<something else>';`) is de-indented to the same level as line 1.

## Steps to Reproduce
see above

## Dialect
tsql

## Version
sqlfluff version 0.6.8, Python 3.10.0

## Configuration
none

**Comment by fdw** [comment]
This might be related:

```sql
create procedure <name>
  @debug bit = 0
as
  begin
    begin try

      declare @a int;
declare @b int;
declare @c int
      select @d = '<something>'
```

Again, I would expect the last two `declare` statements to be on the same indentation level as the previous one. However, for some, reason, some are de-indented. I suspect it has something to do with the semicolons in the lines before.

**Comment by tunetheweb** [comment]
Bad news: currently SQLFluff doesn't support `BEGIN`/`END` so doesn't realise this is a statement block 😞
Good news: It's about to support it thanks to @jpers36 in #1553 ! 🎉

When I test both your SQL with that branch, it no longer breaks the indentation!

**Comment by tunetheweb** [comment]
Added a "Fixes #1555" to the PR so this will be automatically closed once it's merged.

**Comment by fdw** [comment]
Unfortunately, even with the latest version (`0.6.9`), this error exists. One example:

```sql
if @verbose = 1
  begin
    print <something>;
    print <something else>; 
  end
```
is corrected to 
```sql
if @verbose = 1
  begin
    print <something>;
print <something else>;
end
```

**Comment by fdw** [comment]
I just tried the test file again with v0.8.1, and the error still persists.

**Comment by tunetheweb** [comment]
Seems to work fine for me:

```
% more test.sql
-- no semicolons, works
if @verbose = 1
  begin
    print 'something'
    print 'something else'
  end

-- no semicolons, indent is reset
if @verbose = 1
  begin
    print 'something';
    print 'something else';
  end

-- bonus: no semicolons, large indent is not reset
if @verbose = 1
  begin
    print 'something'
            print 'something else'
  end


% sqlfluff fix --dialect tsql --fixed-suffix FIXED -f test.sql
==== finding fixable violations ====
== [test.sql] FAIL                                                                                                                                                                
L:   3 | P:   3 | L003 | Indentation not hanging or a multiple of 4 spaces
L:   6 | P:   3 | L003 | Indentation not hanging or a multiple of 4 spaces
L:  10 | P:   3 | L003 | Indentation not hanging or a multiple of 4 spaces
L:  13 | P:   3 | L003 | Indentation not hanging or a multiple of 4 spaces
L:  17 | P:   3 | L003 | Indentation not hanging or a multiple of 4 spaces
L:  19 | P:  13 | L003 | Indentation not consistent with line #18
L:  20 | P:   3 | L003 | Indentation not hanging or a multiple of 4 spaces
==== fixing violations ====
7 fixable linting violations found
FORCE MODE: Attempting fixes...
Persisting Changes...
== [test.sql] PASS
Done. Please check your files to confirm.

% more testFIXED.sql 
-- no semicolons, works
if @verbose = 1
    begin
        print 'something'
        print 'something else'
    end

-- no semicolons, indent is reset
if @verbose = 1
    begin
        print 'something';
        print 'something else';
    end

-- bonus: no semicolons, large indent is not reset
if @verbose = 1
    begin
        print 'something'
        print 'something else'
    end
```

Can you open a new issue with a minimal, reproducible test case?

**Comment by fdw** [comment]
Sorry, that was my mistake - when I explicitely set the `tsql` dialect, it works as expected. Thank you very much!
